src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,28 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private List<EventListener> listeners = new CopyOnWriteArrayList<>();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
```

Comment:
To enhance the thread-safety and consistency of the `notifyEvent` method, consider iterating over a snapshot of the listener list to protect against concurrent modifications.